src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -428,6 +433,17 @@ public AzureAppConfigurationOptions Map(Func<ConfigurationSetting, ValueTask<Con
             return this;
         }
 
+        /// <summary>
+        /// Configure automatic http failover for Azure App Configuration.
+        /// </summary>
+        /// <param name="enabled">Enable automatic Http failover or not</param>
+        public AzureAppConfigurationOptions EnableAutoFailover(bool enabled)
```

Comment:
If we're going with this public API instead of passing the flag to `Connect` API, I'd suggest `DisableAutoFailover` instead. It looks odd to have an API that enables something which is already enabled by default.

`public AzureAppConfigurationOptions DisableAutoFailover()`

Usage:

```
options.Connect(connStr)
       .Select("App1*")
       .DisableAutoFailover();
```

 cc @jimmyca15

Reply:
This feature is about discovering replica endpoints automatically. It's not about automatic failover. We never need user intervention to failover from one replica to another. I'd suggest we name this feature something like `AutoReplicaDiscovery`.

Yes, I thought we would pass this as a flag to the `Connect(..., autoReplicaDiscovery: true)` API too.

Reply:
I like the suggestion on the feature name.

For how the API should look like,  I think both ways are good to me. Compared with adding a flag in the `Connect()` API,  the only advantage of using an extension method comes to my mind is that it is more flexible for usage, for testing as an example, not have to call `connect()` to enable/disable auto failover first, just call extension method directly is fine.

Reply:
Tests have InternalsVisibleTo the assembly. It shouldn't be a problem.

Reply:
How about calling it `AutoDiscoverReplica`?

`Connect(..., autoDiscoverReplica: true)`

Reply:
As discussed offline, let's move this to a property.